data_steward/validation/ehr_union.py

```diff
@@ -757,7 +757,9 @@ def main(input_dataset_id, output_dataset_id, project_id, hpo_ids_ex=None):
     # Get all hpo_ids.
     hpo_ids = [item['hpo_id'] for item in bq_utils.get_hpo_info()]
     if hpo_ids_ex:
-        hpo_ids = list(set(hpo_ids) - set(hpo_ids_ex))
+        for ex_id in hpo_ids_ex:
+            if ex_id in hpo_ids:
+                hpo_ids.remove(ex_id)
```

Comment:
Nice catch. You can also do list comprehension.
```suggestion
        hpo_ids = [hpo_id for hpo_id in hpo_ids if hpo_id not in hpo_ids_ex]
```

Reply:
that would work too to preserve the order, yes.